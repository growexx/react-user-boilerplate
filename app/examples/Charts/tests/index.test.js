/**
 *
 * Tests for Charts
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
import { browserHistory } from 'react-router-dom';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import history from 'utils/history';
import { ConnectedRouter } from 'connected-react-router';
import Charts from '../index';
import configureStore from '../../../configureStore';

let store;

const props = {};

const componentWrapper = updatedProps =>
  render(
    <Provider store={store}>
      <IntlProvider locale="en">
        <ConnectedRouter history={history}>
          <Charts {...props} {...updatedProps} />
        </ConnectedRouter>
      </IntlProvider>
    </Provider>,
  );

describe('Check component:<Charts /> is rendering properly', () => {
  beforeAll(() => {
    store = configureStore({}, browserHistory);
    jest.useFakeTimers();
  });

  afterEach(() => {
    jest.runOnlyPendingTimers();
    jest.useRealTimers();
  });

  it('Should render and match the snapshot', () => {
    global.Date = jest.fn(() =>
      Object.assign(Date, {
        getTime: jest.fn(() => ({
          toString: jest.fn(() => ({
            slice: jest.fn(() => '000'),
          })),
        })),
      }),
    );
    global.Date.now = jest.fn(() => 1530518207007);
    const {
      container: { firstChild },
    } = componentWrapper();
    expect(firstChild).toMatchSnapshot();
  });
});
