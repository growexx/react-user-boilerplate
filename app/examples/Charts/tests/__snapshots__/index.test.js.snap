// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check component:<Charts /> is rendering properly Should render and match the snapshot 1`] = `
.c0 {
  width: 100%;
  min-width: 325px;
}

.c3 {
  width: 97%;
  margin: 0 auto;
  text-align: center;
}

.c2 {
  width: 50%;
  text-align: center;
}

.c1 {
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
}

@media screen and (max-width:500px) {
  .c0 {
    max-width: 325px;
    min-width: initial;
  }
}

@media all and (max-width:1024px) {
  .c2 {
    width: 100%;
    margin-bottom: 20px;
  }
}

@media all and (min-width:768px) and (max-width:1024px) {
  .c1 .ant-statistic-content {
    font-size: 23px;
  }
}

@media all and (min-width:480px) and (max-width:768px) {
  .c1 .ant-statistic-content {
    font-size: 20px;
  }
}

@media all and (max-width:480px) {
  .c1 .ant-statistic-content {
    font-size: 17px;
  }
}

<div>
  <div
    class="ant-page-header CommonClasses.PageHeader ant-page-header-ghost"
  >
    <div
      class="ant-page-header-heading"
    >
      <div
        class="ant-page-header-heading-left"
      >
        <span
          class="ant-page-header-heading-title"
          title="Dashboard"
        >
          Dashboard
        </span>
      </div>
      <span
        class="ant-page-header-heading-extra"
      >
        <div
          class="CommonClasses.filters"
        >
          <div
            class="ant-space ant-space-horizontal ant-space-align-center CommonClasses.filterItem align-middle m-1"
          >
            <div
              class="ant-space-item"
            >
              <div
                class="ant-select c0 ant-select-single ant-select-show-arrow"
              >
                <div
                  class="ant-select-selector"
                >
                  <span
                    class="ant-select-selection-search"
                  >
                    <input
                      aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                      aria-autocomplete="list"
                      aria-controls="rc_select_TEST_OR_SSR_list"
                      aria-haspopup="listbox"
                      aria-owns="rc_select_TEST_OR_SSR_list"
                      autocomplete="off"
                      class="ant-select-selection-search-input"
                      id="rc_select_TEST_OR_SSR"
                      readonly=""
                      role="combobox"
                      style="opacity: 0;"
                      type="search"
                      unselectable="on"
                      value=""
                    />
                  </span>
                  <span
                    class="ant-select-selection-item"
                    title="Current Month (MTD)"
                  >
                    Current Month (MTD)
                  </span>
                </div>
                <span
                  aria-hidden="true"
                  class="ant-select-arrow"
                  style="user-select: none;"
                  unselectable="on"
                >
                  <span
                    aria-label="down"
                    class="anticon anticon-down ant-select-suffix"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="down"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                      />
                    </svg>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </span>
    </div>
  </div>
  <div>
    <div
      class="ant-row d-flex justify-content-around flex-row w-100"
    >
      <div
        class="ant-space ant-space-horizontal ant-space-align-center d-flex justify-content-around flex-row w-100 c1"
      >
        <div
          class="ant-space-item"
          style="margin-right: 8px;"
        >
          <div
            class="ant-statistic m-4"
          >
            <div
              class="ant-statistic-title"
            >
              New Users
            </div>
            <div
              class="ant-statistic-content"
            >
              <span
                class="ant-statistic-content-value"
              >
                <span
                  class="ant-statistic-content-value-int"
                >
                  
                  000
                </span>
                
              </span>
            </div>
          </div>
        </div>
        <div
          class="ant-space-item"
          style="margin-right: 8px;"
        >
          <div
            class="ant-statistic m-4"
          >
            <div
              class="ant-statistic-title"
            >
              Payouts
            </div>
            <div
              class="ant-statistic-content"
            >
              <span
                class="ant-statistic-content-prefix"
              >
                $
              </span>
              <span
                class="ant-statistic-content-value"
              >
                <span
                  class="ant-statistic-content-value-int"
                >
                  
                  0
                </span>
                <span
                  class="ant-statistic-content-value-decimal"
                >
                  .00
                </span>
              </span>
            </div>
          </div>
        </div>
        <div
          class="ant-space-item"
        >
          <div
            class="ant-statistic m-4"
          >
            <div
              class="ant-statistic-title"
            >
              Earnings
            </div>
            <div
              class="ant-statistic-content"
            >
              <span
                class="ant-statistic-content-prefix"
              >
                $
              </span>
              <span
                class="ant-statistic-content-value"
              >
                <span
                  class="ant-statistic-content-value-int"
                >
                  
                  0
                </span>
                <span
                  class="ant-statistic-content-value-decimal"
                >
                  .00
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-divider ant-divider-horizontal"
      role="separator"
      width="auto"
    />
    <div
      class="ant-row m-1 d-flex justify-content-center flex-row align-items-start"
    >
      <div
        class=" d-flex justify-content-center flex-column c2"
      >
        <h4>
          Sales by Category
        </h4>
        <div
          class="ant-space ant-space-horizontal ant-space-align-center d-flex justify-content-around flex-row m-2"
        >
          <div
            class="ant-space-item"
          >
            <div
              class="ant-skeleton ant-skeleton-element ant-skeleton-active"
            >
              <span
                class="ant-skeleton-button ant-skeleton-button-lg ant-skeleton-button-circle"
                style="width: 250px; height: 250px;"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class=" d-flex justify-content-center flex-column c2"
      >
        <h4>
          Sales by University
        </h4>
        <div
          class="ant-space ant-space-horizontal ant-space-align-center d-flex justify-content-around flex-row m-2"
        >
          <div
            class="ant-space-item"
          >
            <div
              class="ant-skeleton ant-skeleton-element ant-skeleton-active"
            >
              <span
                class="ant-skeleton-button ant-skeleton-button-lg ant-skeleton-button-circle"
                style="width: 250px; height: 250px;"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-row mt-5 d-flex justify-content-center flex-column w-90 c3"
      >
        <h4>
          Sign up
        </h4>
        <div
          class="ant-space ant-space-horizontal ant-space-align-center d-flex justify-content-around flex-row w-100"
        >
          <div
            class="ant-space-item"
          >
            <div
              class="ant-skeleton ant-skeleton-element ant-skeleton-active"
            >
              <span
                class="ant-skeleton-button ant-skeleton-button-lg"
                style="width: 500px; height: 250px;"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
